logstash-filter-tld.gemspec

```
@@ -17,12 +17,14 @@ Gem::Specification.new do |s|
   # Special flag to let us know this is actually a logstash plugin
   s.metadata = { "logstash_plugin" => "true", "logstash_group" => "filter" }
 
+  # Logstash 7.17.x-8.x uses ruby 2.5-3.1 compatiblity
+  # public_suffix 4.x+ includes ruby syntax from 2.3
   # public_suffix 5.x+ includes ruby syntax from 2.6
   s.required_ruby_version = '>= 2.3'
 
   # Gem dependencies
   s.add_runtime_dependency "logstash-core-plugin-api", ">= 1.60", "<= 2.99"
-  s.add_runtime_dependency 'public_suffix', '~> 4'
+  s.add_runtime_dependency 'public_suffix', '> 3', '< 6'
```

My bad, sorry '> 3' would allow 3.x:

```suggestion
  s.add_runtime_dependency 'public_suffix', '> 4', '< 6'
```

```
jruby-9.3.10.0 :007 > Gem::Requirement.new("> 3", "< 6").satisfied_by? Gem::Version.new("3.1.0")
 => true 
jruby-9.3.10.0 :008 > Gem::Requirement.new("> 4", "< 6").satisfied_by? Gem::Version.new("3.1.0")
 => false 
```